src/routes/rgbpp/address.ts

```diff
@@ -200,6 +218,49 @@ const addressRoutes: FastifyPluginCallback<Record<never, never>, Server, ZodType
         xudtBalances[type_hash].pending_amount = BI.from(xudtBalances[type_hash].pending_amount)
           .add(BI.from(amount))
           .toHexString();
+      });
+
+      // Find spent RgbppLock Xudt assets in unconfirmed transactions' inputs
+      const allTxs = await fastify.bitcoin.getAddressTxs({ address: btc_address });
+      const unconfirmedTxids = allTxs.filter((tx) => !tx.status.confirmed).map((tx) => tx.txid);
+      const spendingInputCellsGroup = await Promise.all(
+        unconfirmedTxids.map(async (txid) => {
+          const inputCells = await fastify.transactionProcessor.getPendingInputCellsByTxid(txid);
+          const inputRgbppCells = getRgbppLockCellsByCells(filterCellsByTypeScript(inputCells, typeScript));
+          const inputCellLockArgs = inputRgbppCells.map((cell) => unpackRgbppLockArgs(cell.cellOutput.lock.args));
+
+          const txids = uniq(inputCellLockArgs.map((args) => remove0x(args.btcTxid)));
+          const txs = await Promise.all(txids.map((txid) => fastify.bitcoin.getTx({ txid })));
+          const txsMap = txs.reduce(
+            (sum, tx, index) => {
+              const txid = txids[index];
+              sum[txid] = tx ?? null;
+              return sum;
+            },
+            {} as Record<string, BTCTransaction | null>,
+          );
+
+          return inputRgbppCells.filter((cell, index) => {
+            const lockArgs = inputCellLockArgs[index];
+            const tx = txsMap[remove0x(lockArgs.btcTxid)];
+            const utxo = tx?.vout[lockArgs.outIndex];
+            return utxo?.scriptpubkey_address === btc_address;
+          });
+        }),
+      );
+      const spendingInputCells = spendingInputCellsGroup.flat();
+      const spendingXudtBalances = await fastify.rgbppCollector.getRgbppBalanceByCells(spendingInputCells);
+      Object.values(spendingXudtBalances).forEach(({ amount, type_hash, ...xudtInfo }) => {
+        if (!xudtBalances[type_hash]) {
+          xudtBalances[type_hash] = {
+            ...xudtInfo,
+            type_hash,
+            total_amount: '0x0',
+            available_amount: '0x0',
+            pending_amount: '0x0',
+          };
+        }
+
         xudtBalances[type_hash].total_amount = BI.from(xudtBalances[type_hash].total_amount)
```

Comment:
~~With this step, adding `Unconfirmed Spent Outputs`' xUDT value to total_amount,
the result of the described test in https://github.com/ckb-cell/btc-assets-api/issues/202#issue-2435982176 will be:
```js
  total_amount: '1399',
  available_amount: '0',
  pending_amount: '499',
```
Am I right?~~

No, I am wrong. See https://github.com/ckb-cell/btc-assets-api/blob/95fac7e5f9f7ddf2ed3d0c6b74693fda27732654/src/routes/rgbpp/address.ts#L206-L221
Pending balance is not calculated into total_amount.